src/main/java/seedu/planner/model/ModulePlanner.java

```diff
@@ -447,6 +435,140 @@ private void sortAvailableModules(List<Module> modulesAvailable, UserProfile use
+    }
+
+    /**
+     * Insert the number of IT professionalism modules to the credit mapping..
+     */
+    private void mapItProfessionalism() {
+        int numOfItProfessionalism = countProgrammeRequirementNonBd(
+                                        DegreeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT);
+        statusMap.put(DegreeRequirement.IT_PROFESSIONALISM, new int[] {numOfItProfessionalism});
+    }
+
+    /**
+     * Insert the number of industrial experience requirement modules to the credit mapping.
+     */
+    private void mapIndustrialExperienceRequirement() {
+        int numOfIndExpReq = countProgrammeRequirementNonBd(DegreeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT);
+        statusMap.put(DegreeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT, new int[] {numOfIndExpReq});
+    }
+
+    /**
+     * Count the number of Team Project Modules and insert to the credit mapping.
+     */
+    private void mapTeamProject() {
+        int numOfTeamProject = 0;
+        for (Module m : getAllModulesTaken()) {
+            Optional<ModuleDescription> moduleDescription = getModuleDescription(m.getCode());
+            if (moduleDescription.isPresent()
+                    && moduleDescription.get().getRequirement().equals(DegreeRequirement.BREATH_AND_DEPTH)
+                    && moduleDescription.get().getFocusAreas().isEmpty()) {
+                numOfTeamProject += m.getCreditCount();
+            }
+        }
+        statusMap.put(DegreeRequirement.TEAM_PROJECT, new int[] {numOfTeamProject});
+    }
+
+    /**
+     * Count the number of the user's focus area requirement and insert to the credit mapping.
+     */
+    private void mapFocusAreasRequirement() {
+        List<FocusArea> focusAreas = new ArrayList<>(getUserProfile().getFocusAreas());
+        if (!focusAreas.isEmpty()) {
+            int[] numOfFocusAreasRequirement = new int[focusAreas.size()];
+            for (Module m : getAllModulesTaken()) {
+                Optional<ModuleDescription> moduleDescription = getModuleDescription(m.getCode());
+                for (int i = 0; i < focusAreas.size(); i++) {
+                    if (moduleDescription.isPresent()
+                            && moduleDescription.get().getRequirement().equals(DegreeRequirement.BREATH_AND_DEPTH)
+                            && !moduleDescription.get().getFocusAreas().isEmpty()
+                            && moduleDescription.get().getFocusAreas().get(0).equals(focusAreas.get(i))) {
+                        numOfFocusAreasRequirement[i] += m.getCreditCount();
+                    }
+                }
+            }
+            statusMap.put(DegreeRequirement.FOCUS_AREA_REQUIREMENTS, numOfFocusAreasRequirement);
+        }
+    }
+
+    /**
+     * Maps each requirements to the number of modules fulfilling it in the planner
+     *
+     * @return the mapping
+     */
+    public ObservableMap<DegreeRequirement, int[]> status() {
```

Comment:
Is the return value used?
Would also be good to rename to something like `setStatus` or something with a verb

Reply:
Yes it's used

Reply:
Can we merge after resolve? cause i don't know what to add for the test lol